src/context/directory/handlers/pages.ts

```diff
@@ -6,33 +6,41 @@ import log from '../../../logger';
 import {
   getFiles, existsMustBeDir, dumpJSON, loadJSON
 } from '../../../utils';
+import { DirectoryHandler } from '.'
 
-function parse(context) {
+type ParsedPages = {
+  pages: unknown[] | undefined
+}
+
+function parse(context): ParsedPages {
   const pagesFolder = path.join(context.filePath, constants.PAGES_DIRECTORY);
   if (!existsMustBeDir(pagesFolder)) return { pages: undefined }; // Skip
 
-  const files = getFiles(pagesFolder, [ '.json', '.html' ]);
+  const files: string[] = getFiles(pagesFolder, ['.json', '.html']);
 
-  const sorted = {};
-
-  files.forEach((file) => {
+  const sorted: {
+    [key: string]: {
+      meta?: string
+      html?: string
+    }
+  } = files.reduce((acc, file) => {
     const { ext, name } = path.parse(file);
-    if (!sorted[name]) sorted[name] = {};
-    if (ext === '.json') sorted[name].meta = file;
-    if (ext === '.html') sorted[name].html = file;
-  });
+    if (!acc[name]) acc[name] = {};
+    if (ext === '.json') acc[name].meta = file;
+    if (ext === '.html') acc[name].html = file;
+    return acc
+  }, {});
 
-  const pages = [];
-  Object.values(sorted).forEach((data) => {
+  const pages = Object.values(sorted).map((data) => {
```

Comment:
`pages` may have undefined items in it now that you're using map

Reply:
Good point, though I walked through and the `sorted` value is the result of a reduce, which at worse is an empty object. `Object.values({})` evaluates to an empty array, so I don't believe it will ever be undefined.

However, while doing so, I realized that the map function doesn't always return a value, so I've instead swapped it with `flatMap` to weed out any potentially undefined values it generates.